### PR TITLE
UCT/API: Make uct_worker_progress() inline.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4006/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls review

**Comment by yosefe** [comment]
@bbenton pls take a look as well (api is affected)

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -1362,6 +1347,24 @@ ucs_status_t uct_rkey_release(const uct_rkey_bundle_t *rkey_ob);
...
+ *
+ * @param [in]  worker        Handle to worker.
+ */
+UCT_INLINE_API void uct_worker_progress(uct_worker_h worker)
+{
+    ucs_callbackq_dispatch(&worker->progress_q);
+}
+
+
+/**
```
Comment (shamisp) [inline src/uct/api/uct.h]: What is the reason for code moving ?

File: src/uct/api/tl.h [hunk 2]
```diff
@@ -204,6 +205,14 @@ typedef struct uct_iface_ops {
 
 
 /**
+ * Progress engine
```
Comment (shamisp) [inline src/uct/api/tl.h]: A bit more description here about the struct will help

**Comment by shamisp** [comment]
@MattBBaker this change involves ugni modification. Please take a look/test.

File: src/uct/api/uct.h (hunk 1 above)
Comment (yosefe) [inline src/uct/api/uct.h]: because all inline functions are in the end of the file

**Comment by shamisp** [comment]
It also may impact ROCM patch. I want to get ROCM first, it has been pending for a while ...

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4031/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4036/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@sergsagal1 - does it break ROCM ? can you please test. thanks

**Comment by MattBBaker** [comment]
Improved performance, but getting warnings about missing ucp_requests
```
[1499371192.222980] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015e9c0 was not returned to mpool ucp_requests
[1499371192.223018] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015eac0 was not returned to mpool ucp_requests
[1499371192.223021] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015ebc0 was not returned to mpool ucp_requests
[1499371192.223024] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015ecc0 was not returned to mpool ucp_requests
[1499371192.223026] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015edc0 was not returned to mpool ucp_requests
[1499371192.223028] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015eec0 was not returned to mpool ucp_requests
[1499371192.223030] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015efc0 was not returned to mpool ucp_requests
[1499371192.223033] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f0c0 was not returned to mpool ucp_requests
[1499371192.223035] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f1c0 was not returned to mpool ucp_requests
[1499371192.223037] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f2c0 was not returned to mpool ucp_requests
[1499371192.223039] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f3c0 was not returned to mpool ucp_requests
[1499371192.223042] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f4c0 was not returned to mpool ucp_requests
[1499371192.223044] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f5c0 was not returned to mpool ucp_requests
[1499371192.223046] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f6c0 was not returned to mpool ucp_requests
[1499371192.223049] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f7c0 was not returned to mpool ucp_requests
[1499371192.223051] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f8c0 was not returned to mpool ucp_requests
[1499371192.223055] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015f9c0 was not returned to mpool ucp_requests
[1499371192.223059] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015fac0 was not returned to mpool ucp_requests
[1499371192.223063] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015fbc0 was not returned to mpool ucp_requests
[1499371192.223066] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015fcc0 was not returned to mpool ucp_requests
[1499371192.223070] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015fdc0 was not returned to mpool ucp_requests
[1499371192.223078] [nid00004:27345:0]          mpool.c:38   UCX  WARN  object 0x2aaab015fec0 was not returned to mpool ucp_requests
```

**Comment by yosefe** [comment]
@MattBBaker can you pls review the ugni changes in this patch? i've probably misses something

**Comment by yosefe** [comment]
@shamisp @sergsagal1 i've checked that UCX compiles with RoCM with this patch

**Comment by MattBBaker** [comment]
@yosefe The UGNI portions look good. And the error seems intermittent. I just did several more runs and didn't get the failure again.

**Comment by bbenton** [comment]
:+1:  for Doc/API

**Pull request merged**